doc/api/api_changes.rst

```diff
@@ -11,6 +11,19 @@ sources of the changes you are experiencing.
 For new features that were added to matplotlib, please see
 :ref:`whats-new`.
 
+Changes in 2.1.0
+================
+
+Code changes
```

Comment:
I'd rather close this PR and make `margin` a tristate as discussed above.  In fact, making `autoscale` a contextmanager is a bit misleading given that at the exit of the contextmanager (assuming it temporarily disabled autoscaling), all the artists, *including those created while the cm was active*, will participate in the next autoscaling... which defeats the purpose of the cm IMO.  Thoughts?

The fix for z autoscaling should go in its own PR.